### ISSUE TITLE
Zerovalent error on assert_modifiable

### ISSUE THREAD

**Issue by fubuloubu** [vyperlang/vyper#1823]
### Version Information

* vyper Version (output of `vyper --version`): 0.1.0b15
* OS: linux
* Python Version (output of `python --version`): 3.7.5

### What's your issue about?
When compiling this contract:
```python
from vyper.interfaces import ERC721 
 
@public 
def foo(): 
    assert_modifiable(ERC721(msg.sender).safeTransferFrom(msg.sender, self, 1, b""))
```
Vyper raises this error:
```bash
File "vyper/parser/stmt.py", line 89, in __init__
    self.lll_node = self.stmt_table[stmt_type]()
  File "vyper/parser/stmt.py", line 94, in expr
    return Stmt(self.stmt.value, self.context).lll_node
  File "vyper/parser/stmt.py", line 89, in __init__
    self.lll_node = self.stmt_table[stmt_type]()
  File "vyper/parser/stmt.py", line 390, in call
    return STMT_DISPATCH_TABLE[self.stmt.func.id](self.stmt, self.context)
  File "vyper/functions/signatures.py", line 135, in g
    context,
  File "vyper/functions/signatures.py", line 84, in process_arg
    vsub = vsub or Expr.parse_value_expr(arg, context)
  File "vyper/parser/expr.py", line 1157, in parse_value_expr
    return unwrap_location(cls(expr, context).lll_node)
  File "vyper/parser/parser_utils.py", line 467, in unwrap_location
    return LLLnode.from_list(['mload', orig], typ=orig.typ)
  File "vyper/parser/lll_node.py", line 379, in from_list
    valency=valency,
  File "vyper/parser/lll_node.py", line 120, in __init__
    "Can't have a zerovalent argument to an opcode or a pseudo-opcode! "
CompilerPanic: Can't have a zerovalent argument to an opcode or a pseudo-opcode! seq: [seq,
  [assert, [extcodesize, [sload, 1 <self.token>]]],
  [assert, [ne, address, [sload, 1 <self.token>]]],
  [assert,
    [call,
      gas,
      [sload, 1 <self.token>],
      0,
      [with,
        _poz,
        128,
        [seq,
          [mstore, 384, 3096268766],
          [mstore, 416, caller],
          [mstore, 448, address],
          [mstore, 480, [calldataload, 4 <_tokenId>]],
          [mstore, 512, _poz],
          [with,
            _s,
            /* Create bytes[0]: b'' */ [seq, [mstore, 320, 0], 320],
            [seq,
              /* Memory copy */ 
              [with,
                _source,
                _s,
                [with,
                  _sz,
                  [add, 32, [mload, _source]],
                  [assert, [call, gas, 4, 0, _source, _sz, [add, 416, _poz], _sz]]]],
              [set, _poz, [add, 32, [ceil32, [add, _poz, [mload, _s]]]]]]],
          412]],
      1188,
      0,
      0]]]. Please file a bug report. Please create an issue.
```

**Comment by charles-cooper** [comment]
The error message is confusing but the problem is that `safeTransferFrom` doesn't return anything, so there is nothing to assert.

**Comment by fubuloubu** [comment]
Yeah, it shouldn't compile but the error message is off.

**Comment by iamdefinitelyahuman** [comment]
This is solved by #2029
